### PR TITLE
Missing Unit Tests

### PR DESCRIPTION
**Pull request by Diogossilva03**

### CODE REVIEW

File: utest/test_visualise_abstractgraph.py [hunk 1]
```diff
@@ -3,14 +3,26 @@
...
+    from robotmbt.visualise.graphs.scenariostategraph import ScenarioStateGraph
     from robotmbt.visualise.models import *
 
     VISUALISE = True
 except ImportError:
     VISUALISE = False
 
+
 if VISUALISE:
     class TestVisualiseAbstractGraph(unittest.TestCase):
```
Comment (Egietje) [inline utest/test_visualise_abstractgraph.py]: It seems you have essentially re-added the unit tests that were deleted because they are more like acceptance tests.

File: utest/test_visualise_models.py [hunk 2]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            self.assertEqual(result, long_word)
+        
+        def test_split_name_two_words_short(self):
+            """Test two short words that fit on one line."""
+            result = ScenarioInfo._split_name("Hello World")
+            self.assertEqual(result, "Hello World")
+        
+        def test_split_name_two_words_exceeds_limit(self):
+            """Test two words that together exceed 20 characters."""
+            result = ScenarioInfo._split_name("HelloSupercalifragilisticexpialidocious")
```
Comment (Egietje) [inline utest/test_visualise_models.py]: You don´t have two words here

File: utest/test_visualise_models.py [hunk 3]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            self.assertEqual(result, "This is a short name")
+        
+        def test_split_name_multiple_words_need_split(self):
+            """Test multiple words that need to be split."""
+            name = "This is a very long scenario name that should be split"
+            result = ScenarioInfo._split_name(name)
+            # Should be split into multiple lines
+            self.assertIn("\n", result)
+            lines = result.split("\n")
+            self.assertGreater(len(lines), 1)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: You are performing the same check twice
How about checking that the output is equal to the input in terms of words (but not white-space)?

File: utest/test_visualise_models.py [hunk 4]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+        
+        def test_split_name_with_long_word_adjusts_length(self):
+            """Test name with extremely long word (45 chars) that exceeds default desired length."""
+            name = "NormalWord ExtremelyLongWordThatExceedsTwentyCharacters AnotherWord"
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("ExtremelyLongWordThatExceedsTwentyCharacters", result)
+            result_words = result.replace('\n', ' ').split()
```
Comment (Egietje) [inline utest/test_visualise_models.py]: What are you testing here? You're undoing any newlines added and checking that it's the same as the input? That's a fine test on its own, but why the split? And that doesn't seem like the objective of the test (given its name)

File: utest/test_visualise_models.py [hunk 5]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            result = ScenarioInfo._split_name(name)
+            self.assertEqual(result, "This is a scenario\nname test")
+        
+        def test_split_name_exact_desired_length(self):
+            """Test two 10-character words where the total length (21) exceeds desired length (20)."""
+            name = "0123456789 0123456789"  # Two 10-character words with space
+            result = ScenarioInfo._split_name(name)
+            self.assertEqual(result, "0123456789 0123456789")
+        
+        def test_split_name_with_trailing_newline_character(self):
```
Comment (Egietje) [inline utest/test_visualise_models.py]: This test misses its point entirely, it just checks that a newline is in the output? It doesn´t even have an input line that should be split?

File: utest/test_visualise_models.py [hunk 6]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            """Test that newline characters in input are handled."""
+            name = "First line\nSecond line"
+            result = ScenarioInfo._split_name(name)
+            # The method splits on spaces, not existing newlines
+            # Existing newlines will be treated as part of words
+            self.assertIn("\n", result)
+        
+        def test_split_name_special_characters(self):
```
Comment (Egietje) [inline utest/test_visualise_models.py]: What does this test exactly? The assert tests for a single phrase being in the result, do we not care about the rest of the input?

File: utest/test_visualise_models.py [hunk 7]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            self.assertEqual(len(words), 4)
+        
+        def test_split_name_very_long_sentence(self):
+            """Test a long Latin placeholder text sentence gets properly split."""
+            name = ("Lorem ipsum dolor sit amet consectetur adipiscing elit sed do "
+                    "eiusmod tempor incididunt ut labore et dolore magna aliqua")
+            result = ScenarioInfo._split_name(name)
+            lines = result.split("\n")
+            for line in lines:
+                self.assertTrue(len(line) > 0)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Why only that they're not empty?

File: utest/test_visualise_models.py [hunk 8]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            result_words = result.replace('\n', ' ').split()
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Why split???

File: utest/test_visualise_models.py [hunk 9]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+        def test_split_name_leading_trailing_spaces(self):
+            """Test name with spaces - algorithm handles word separation."""
+            name = "  Leading and trailing spaces  "
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Leading", result)
+            self.assertIn("trailing", result)
+            self.assertIn("spaces", result)
+            name = "Leading and trailing spaces"
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Leading", result)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: ??? why another assert without the leading/trailing spaces??? Why just test if Leading is in result??

File: utest/test_visualise_models.py [hunk 10]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Pokémon", result)
+        
+        def test_split_name_leading_trailing_spaces(self):
+            """Test name with spaces - algorithm handles word separation."""
+            name = "  Leading and trailing spaces  "
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Leading", result)
+            self.assertIn("trailing", result)
+            self.assertIn("spaces", result)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Are these asserts all we care about here??

File: utest/test_visualise_models.py [hunk 11]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            self.assertEqual(result, expected)
+        
+        def test_split_name_no_spaces(self):
+            """Test a name with no spaces (should not split)."""
+            name = "ThisIsAReallyLongNameWithoutAnySpacesAtAll"
+            result = ScenarioInfo._split_name(name)
+            self.assertEqual(result, name)
+        
+        def test_split_name_one_word_per_line(self):
+            """Test many single-letter words get grouped into reasonable lines."""
```
Comment (Egietje) [inline utest/test_visualise_models.py]: This is two tests in one?
result without whitespace = input without whitespace
lines are all < 25

File: utest/test_visualise_models.py [hunk 12]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            name = "A B C D E F G H I J K L M N O P Q R S T U V W X Y Z"
+            result = ScenarioInfo._split_name(name)
+            lines = result.split("\n")
+            result_letters = result.replace('\n', ' ').replace(' ', '')
+            expected_letters = name.replace(' ', '')
+            self.assertEqual(result_letters, expected_letters)
+            for line in lines:
+                self.assertLessEqual(len(line), 25)
+
+        def test_split_name_returns_string(self):
```
Comment (Egietje) [inline utest/test_visualise_models.py]: We're doing type checking as unit tests now?

File: utest/test_visualise_models.py [hunk 13]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            self.assertIn("Leading", result)
+        
+        def test_split_name_algorithm_correctness(self):
+            """Test specific example showing algorithm's splitting behavior."""
+            name = "This is a test scenario name example"
+            result = ScenarioInfo._split_name(name)
+            expected = "This is a test scenario\nname example"
+            self.assertEqual(result, expected)
+        
+        def test_split_name_no_spaces(self):
```
Comment (Egietje) [inline utest/test_visualise_models.py]: You already have a test for this

File: utest/test_visualise_models.py [hunk 14]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            lines = result.split("\n")
+            for line in lines:
+                self.assertTrue(len(line) > 0)
+            result_words = result.replace('\n', ' ').split()
+            expected_words = name.split()
+            self.assertEqual(result_words, expected_words)
+        
+        def test_split_name_edge_case_previous_line_empty(self):
+            """Test edge case where previous line becomes empty string."""
+            # This tests the condition: if line == '\n'
```
Comment (Egietje) [inline utest/test_visualise_models.py]: No it doesn't??

File: utest/test_visualise_models.py [hunk 15]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            name = "A " * 15
+            result = ScenarioInfo._split_name(name)
+            self.assertIsInstance(result, str)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: You're just doing another string type check?

File: utest/test_visualise_models.py [hunk 16]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+        
+        def test_split_name_unicode_characters(self):
+            """Test splitting names with Unicode characters."""
+            name = "Café naïve résumé Pokémon"
```
Comment (Egietje) [inline utest/test_visualise_models.py]: This only tests two simple unicode characters: é and ï, how about emojis? More difficult things?

File: utest/test_visualise_models.py [hunk 17]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Pokémon", result)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Is this assert really all that would check this case is handled properly?

File: utest/test_visualise_models.py [hunk 18]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            name = "  Leading and trailing spaces  "
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Leading", result)
+            self.assertIn("trailing", result)
+            self.assertIn("spaces", result)
+            name = "Leading and trailing spaces"
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("Leading", result)
+        
+        def test_split_name_algorithm_correctness(self):
```
Comment (Egietje) [inline utest/test_visualise_models.py]: This is essentially equal to another test you already have

**Review by Egietje — CHANGES_REQUESTED** [review]

File: utest/test_visualise_models.py [hunk 19]
```diff
@@ -29,6 +29,177 @@ def test_scenarioInfo_Scenario(self):
...
+            name = "Test with-dash_and_underscore plus@symbol"
+            result = ScenarioInfo._split_name(name)
+            self.assertIn("with-dash_and_underscore", result)
+        
+        def test_split_name_multiple_spaces(self):
+            """Test that multiple spaces are collapsed to single splits."""
+            name = "Test    with    multiple    spaces"
+            result = ScenarioInfo._split_name(name)
+            # Multiple spaces should be treated as single spaces for splitting
+            words = [w for w in result.replace("\n", " ").split(" ") if w]
```
Comment (Egietje) [inline utest/test_visualise_models.py]: This won't tell you if multiple spaces are collapsed into a single split...

**Review by Egietje — CHANGES_REQUESTED** [review]

File: utest/test_visualise_abstractgraph.py (hunk 1 above)
Comment (Egietje) [reply utest/test_visualise_abstractgraph.py]: Turns out they weren't deleted on main yet - in that case these are just straight copies

File: utest/test_visualise_models.py [hunk 20]
```diff
@@ -29,6 +29,108 @@ def test_scenarioInfo_Scenario(self):
...
+            result = ScenarioInfo._split_name(name)
+            # First word sets desired length to 34, second word fits
+            self.assertEqual(result, "Supercalifragilistic\nHello")
+
+        def test_split_name_multiple_words_need_split(self):
+            """Test multiple words that need to be split."""
+            name = "This is a very long scenario name that should be split"
+            result = ScenarioInfo._split_name(name)
+            # Check that output preserves all words in order
+            result_words = result.replace('\n', ' ').split()
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Why split?

File: utest/test_visualise_models.py (hunk 4 above)
Comment (Egietje) [reply utest/test_visualise_models.py]: Still has split

File: utest/test_visualise_models.py [hunk 21]
```diff
@@ -29,6 +29,108 @@ def test_scenarioInfo_Scenario(self):
...
+            result = ScenarioInfo._split_name(name)
+            # Algorithm keeps them together because 21 is closer to 20 than 10 is
+            self.assertEqual(result, "0123456789 0123456789")
+
+        def test_split_name_special_characters(self):
+            """Test splitting names with special characters."""
+            name = "Test with-dash_and_underscore plus@symbol Café naïve résumé Pokémon 🚀🎉"
+            result = ScenarioInfo._split_name(name)
+            # All words should be preserved
+            result_words = result.replace('\n', ' ').split()
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Why split?

File: utest/test_visualise_models.py [hunk 22]
```diff
@@ -29,6 +29,108 @@ def test_scenarioInfo_Scenario(self):
...
+            expected_words = name.split()
+            self.assertEqual(result_words, expected_words)
+        
+        def test_split_name_leading_trailing_spaces(self):
+            """Test handling of leading/trailing spaces."""
+            name = "  Leading and trailing spaces  "
+            result = ScenarioInfo._split_name(name)
+            # The algorithm preserves leading/trailing spaces as empty strings
+            result_words = result.replace('\n', ' ').split()
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Why split?

File: utest/test_visualise_models.py [hunk 23]
```diff
@@ -29,6 +29,108 @@ def test_scenarioInfo_Scenario(self):
...
+            expected_words = name.split()
+            self.assertEqual(result_words, expected_words)
```
Comment (Egietje) [inline utest/test_visualise_models.py]: Should they be equal? What is the expected behavior for leading/trailing spaces?

**Review by Egietje — CHANGES_REQUESTED** [review]

**Comment by JWillegers** [comment]
Merged as https://github.com/MASTERS-Y2Q1-ISEP/robotframeworkMBT/pull/64 due to rebasing
